README.md

```diff
@@ -5,3 +5,415 @@ An implementation of [Soutei](http://okmij.org/ftp/papers/Soutei.pdf).
+accepts a collection of facts as, effectively, parameters that are made available to our rules
+through the `application` assertion.  (NB: I'm doing my best to distinguish Soutei-the-language from
+`avaleryar`-the-implementation-of-Soutei-the-language, I hope this isn't too confusing).  So to
+demonstrate, let's write a needlessly complicated rule determining whether the dancing we can do is,
+properly a Safety Dance.
+
+```prolog
+can(?activity) :-
+  want-to(?activity).
+
+can(sing).
+
+;; we can dance if we want to
+want-to(dance).
+
+;; we can act if we want to
+want-to(act).
+
+;; we only want to go when the night is young, and so am and I
+want-to(go) :-
+  clock says time-of(night, young),
+  bio says age-of(me, young).
+
+;; we can overextend the efficacy a questionable pop-culture reference
+safety(?activity) :-
+  can(?activity),
+  application says out-of(control, everything),
+  application says doing-it(from, pole),
+  application says doing-it(to, pole),
+  application says looking-at(hands, ?somebody),
+  ?somebody says taking(the-chance).
+```
+
+This (completely inane, it's getting pretty late as I write this---the examples section below won't
+be this silly) new rule uses a bunch of information provided by the application querying for
+authorization advice (`application says ...`)  in addition to some locally written rules
+(`can(?activity)`).  It also uses an assertion _determined by the query_ (`?somebody says
+taking(the-chance)`) to ultimately establish that indeed, `safety(dance)`.  The ability to
+dynamically choose different assertions in which to reason is a powerfully expressive feature of
+Soutei.
+
+TODO: unification and native predicates.
+
+TODO: why encoding "'Cause your friends don't dance, and if they don't dance, then they're no
+friends of mine" isn't (naively) possible.
+
+TODO: mode checking.
+
+TODO: monotonicity.
+
+## Examples and Advice
+
+### Example: Unix File Permissions
+
+Here is a simplified version of file permissions on Unix.  We assume that the application will tell
+us what the file is, that there's an assertion named after the file that knows what permissions are
+set on it, who owns it, and which group it's associated with, as well as that each group has an
+assertion named after it that can tell us whether a user is a member of that group.  For example, we
+might have an assertion for some file that looked like:
+
+```prolog
+;; assertion for /path/to/some/file
+
+owner(mary).
+group(wheel).
+perm(user, read).
+perm(user, write).
+perm(user, execute).
+perm(group, read).
+perm(group, execute).
+perm(other, read).
+```
+
+This would represent a file owned by `mary`, with group `wheel`, with permissions `754` (as in,
+`chmod 754 /path/to/some/file`).
+
+```prolog
+;; allow ?access when the user is the owner and ?access is enabled for them.
+may(?access) :-
+  application says file(?file),
+  application says user(?user),
+  ?file says owner(?user),
+  ?file says perm(user, ?access).
```

Comment:
Maybe a comment here that this is the symbol `user` not the variable `?user`.  At first I suspected a typo, but I think you're right to use `user` as the symbol for `perm` since that's what `chmod` calls it.

Reply:
Heh, I was worried about the same thing---the string "user" appears with three different meanings in the rule.  I'll add a note on the next push.